### ISSUE TITLE
如何解决多线程下多个样本的输入冲突问题？

### ISSUE THREAD

**Issue by zbhoscar** [FesianXu/Parallel-DataLoader-in-TensorFlow#1]
我看了你的代码，有一个疑问：比如我有100个视频，命名为'1.avi','2.avi',...,'100.avi'，数据生成器在多线程的情况下，如何保证每个都只生成一遍呢？或者，重复的多遍？

**Comment by FesianXu** [comment]
Hi zbhoscar ! 
I don't know whether i misunderstand you or not. In my understanding, you also need to maintain a global queue to save the name list of your videos. The name list should be shuffled after one training epoch. It may looks like [5.avi, 10.avi, 63,avi ... 3.avi] and the amount is exactly 100. And now your multi-threads data loaders can fetch the videos from your disk by the index of the name in your shuffled list one by one. The multi-threads data loaders should be exclusive in this way.(Also, you need to maintain an global anchor to locate the videos' name which have been loaded.)
Hope it's helpful.

**Comment by FesianXu** [comment]
@zbhoscar

**Comment by zbhoscar** [comment]
Yes, that's exactly the issue I'm facing. But how to make this global anchor? It's bothering me for quite a long time and I am really appreciated for further details!

**Comment by FesianXu** [comment]
@zbhoscar 
The global anchor can be used to mark the the videos location you have already loaded in your global name list. just using a int32 is enough. For example, you can maintain a point `here` begin from 0, meaning that you have not loaded anything.  your each multi-threads data loader should load the data begin from `here` with `batch_size`  and add the number `batch_size` to `here` to update it. Once `here` is larger than `len(your_data_set)`, begin a new epoch or end your training. That is it.

**Comment by zbhoscar** [comment]
@FesianXu Thanks. It is very worth trying.
